mm2src/coins/eth/eth_swap_v2.rs

```diff
@@ -0,0 +1,159 @@
+use super::eth::{addr_from_raw_pubkey, gas_limit, wei_from_big_decimal, EthCoin, EthCoinType, SignedEthTx,
+                 TAKER_SWAP_V2};
+use super::{SendTakerFundingArgs, Transaction, TransactionErr};
+use enum_derives::EnumFromStringify;
+use ethabi::Token;
+use ethcore_transaction::Action;
+use ethereum_types::{Address, U256};
+use futures::compat::Future01CompatExt;
+use std::convert::TryInto;
+
+struct TakerFundingArgs<'a> {
+    dex_fee: U256,
+    payment_amount: U256,
+    maker_address: Address,
+    taker_secret_hash: &'a [u8],
+    maker_secret_hash: &'a [u8],
+    funding_time_lock: u32,
+    payment_time_lock: u32,
+}
+
+impl EthCoin {
+    pub(crate) async fn send_taker_funding_impl(
+        &self,
+        args: SendTakerFundingArgs<'_>,
+    ) -> Result<SignedEthTx, TransactionErr> {
+        let taker_swap_v2_contract = self
+            .swap_v2_contracts
+            .as_ref()
+            .map(|contracts| contracts.taker_swap_v2_contract)
+            .ok_or_else(|| TransactionErr::Plain(ERRL!("Expected swap_v2_contracts to be Some, but found None")))?;
+        let dex_fee = try_tx_s!(wei_from_big_decimal(&args.dex_fee.fee_amount().into(), self.decimals));
+
+        let payment_amount = try_tx_s!(wei_from_big_decimal(
+            &(args.trading_amount + args.premium_amount),
+            self.decimals
+        ));
```

Comment:
Shouldn't we include the dex fee in the total amount to be sent?

P.S. this is how it's done for utxo
https://github.com/KomodoPlatform/komodo-defi-framework/blob/40649999a7644f41b4b7b2924a3b9613051c829a/mm2src/coins/utxo/utxo_common.rs#L4791

Reply:
Hmm I see, as in ethTakerPayment we subtract dexfee from msg.value, then we should add dex_fee to total amount, when we send `EthCoinType::Eth` case.
https://github.com/KomodoPlatform/etomic-swap/blob/5e15641cbf41766cd5b37b4d71842c270773f788/contracts/EtomicSwapTakerV2.sol#L63

But why my send-refund tests passed then?

Reply:
> Hmm I see, as in ethTakerPayment we subtract dexfee from msg.value, then we should add dex_fee to total amount, when we send EthCoinType::Eth case.
> In erc20 case we send U256::from(0) so no need to add dexfee tot total payment.

Yeah, only needed for Eth / platform coin case apparently. Also, for taker approval, we send dex fee separate from amount for both eth / platform coin and tokens
```js
    function takerPaymentApprove(
        bytes32 id,
        uint256 amount,
        uint256 dexFee,
        address maker,
        bytes32 takerSecretHash,
        bytes32 makerSecretHash,
        address tokenAddress
    ) external {
```

> But why my send-refund tests for eth passed then?

can you point me to this test so that I can give it a check as well.

Reply:
> can you point me to this test so that I can give it a check as well.

Its from second pr https://github.com/KomodoPlatform/komodo-defi-framework/pull/2169/files#diff-4bfd82ccb22215a7516f1ac3f5d1913588bee7d70ed917a3f3e5b6309b5a234a

such tests were provided
`send_and_refund_taker_funding_by_secret_eth`
`send_and_refund_taker_funding_by_secret_erc20`
`send_and_refund_taker_funding_timelock_eth`
`send_and_refund_taker_funding_timelock_erc20`

I understood why they were passing, its bcz I didnt add `wait_for_confirmations` after calling refund methods.

I have `Waiting for confirmations… Failed.` issues right now using Geth node for eth and erc20
https://github.com/KomodoPlatform/komodo-defi-framework/actions/runs/10057619461/job/27798973002#step:6:1513

Refund test of `Erc20` token fails confirmation waiting right on `Taker sent ERC20 funding` stage. ("approve" method is used)
ETH refund also has confirmation issue, but `Taker sent ETH funding` tx is confirmed, however next `Taker refunded ETH funding` txz confirmation failing.

Right now Im working on sepolia deploy to see how swaps will work in this environment

Reply:
Yep, we should include dex fee to total payment for eth, will fix it.

@shamardy I understood what was the problem of failing confirmation of eth refund tests. there was a mistake in swap id code line, so tx was reverted by swap contract with `Invalid payment state. Must be PaymentSen` error https://sepolia.etherscan.io/tx/0x423762b259fbbd5667a0cc38a53a52fef062cc15289edc4a87b4dc353851b3b8
I think I was confused by my previous experience of failing nft tests, when in Geth confirmations were failing, while they were working on Sepolia.

Now I can confirm that eth refund methods works (on sepolia)
https://github.com/KomodoPlatform/komodo-defi-framework/actions/runs/10075604842/job/27854240559?pr=2169#step:6:1398
```
test docker_tests::eth_docker_tests::send_and_refund_taker_funding_by_secret_eth ... ok
test docker_tests::eth_docker_tests::send_and_refund_taker_funding_timelock_eth ... ok
```

I will keep working on Sepolia, as it provides the info why tx was reverted.